### ISSUE TITLE
datadog_monitor requires `thresholds` attribute on all monitor types, but API says it only applies to `service check`

### ISSUE THREAD

**Issue by stephenchu** [hashicorp/terraform#9026]
### Terraform Version

v0.7.4
### Affected Resource(s)
- datadog_monitor
### Terraform Configuration Files

``` hcl
resource "datadog_monitor" "ssl_expiration" {
  name    = "foo"
  type    = "event alert"
  query   = "events('status:warn,error tags:myapp,ssl_expiration).rollup('count').last('24h') > 0"
  message = "my message"

  no_data_timeframe = 1440
}
```
### Expected Behavior

I should NOT need to specify attribute `thresholds { ... }` on any monitor type except `service check` type. See Datadog API [#monitor-create](http://docs.datadoghq.com/api/?lang=console#monitor-create), look for "SERVICE CHECK OPTIONS".
### Actual Behavior

During `terraform plan`:

```
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * datadog_monitor.ssl_expiration: "thresholds": required field is not set
```
### Workaround

Add in a `threshold {}` without inner attributes seem to be able to trick Terraform to apply the monitor creation. Although curling the actual monitor API I see an additional `thresholds` attribute show up:

`curl -s -G https://app.datadoghq.com/api/v1/monitor/xxxxxxx -d "......."`

``` json
.
.
.
"overall_state": "No Data",
"type": "event alert",
"options": {
  "notify_audit": false,
  "locked": false,
  "silenced": {},
  "no_data_timeframe": 1440,
  "new_host_delay": 300,
  "notify_no_data": true,
  "thresholds": {}
}
```

**Comment by ojongerius** [comment]
Good catch! We can make it optional and update the documentation. #10526 makes `thresholds` optional so you won't have to use your workaround, and updates the documentation.

**Comment by stack72** [comment]
Closed via #10526
